File: week1/atakantekatan/167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers)-1
        while numbers[left] + numbers[right] != target:
            while numbers[left] + numbers[right] > target:
                right -= 1
            while numbers[left] + numbers[right] < target:
                left += 1
        return [left+1,right+1]

File: week1/alfazya021/167-TwoSumII-InputArrayIsSorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1;
        
        while(left < right) {
            if(target - numbers[left] < numbers[right]) {
                right--;
            } else if(target - numbers[left] > numbers[right]) {
                left++;
            } else {
                break;
            }
        }
        
        int[] answer = {left+1, right+1};
        
        return answer;
    }
}